tls/src/params.cc

```diff
@@ -217,7 +229,8 @@ void params::validate_cert(gnutls_session_t session) {
   if (!_ca.empty()) {
     int ret;
     uint32_t status;
-    ret = gnutls_certificate_verify_peers2(session, &status);
+    ret = gnutls_certificate_verify_peers3(session, _tls_hostname.c_str(),
+                                           &status);
```

Comment:
As in the previous PR, I think this code contains a bug. If we want to use the hostname and we don't provide it as tls_hostname, it will fail.